File: Daily Challanges/2026-03-March/2026-03-04.py
def card_values(cards):
    numeric_values = []
    for card in cards:
        if card[0] == "A":
            numeric_values.append(1)
        elif card[0] in "JKQ" or card[0:2] == "10":
            numeric_values.append(10)
        else:
            numeric_values.append(int(card[0]))

    return numeric_values

test1 = card_values(["3H", "4D", "5S"])
test2 = card_values(["AS", "10S", "10H", "6D", "7D"])
test3 = card_values(["8D", "QS", "2H", "JC", "9C"])
test4 = card_values(["AS", "KS"])
test5 = card_values(["10H", "JH", "QH", "KH", "AH"])

check1 = [3, 4, 5]
check2 = [1, 10, 10, 6, 7]
check3 = [8, 10, 2, 10, 9]
check4 = [1, 10]
check5 = [10, 10, 10, 10, 1]

print(f"Test1: {test1 == check1}")
print(f"Test2: {test2 == check2}")
print(f"Test3: {test3 == check3}")
print(f"Test4: {test4 == check4}")
print(f"Test5: {test5 == check5}")

File: Daily Challanges/2026-03-March/2026-03-04.js
function cardValues(cards) {
  let numericValues = [];
  cards.forEach((card) => {
    const value = card.slice(0, -1);

    if (["J", "Q", "K"].includes(value)) {
      numericValues.push(10);
    } else if (value === "A") {
      numericValues.push(1);
    } else {
      numericValues.push(Number(value));
    }
  });
  return numericValues;
}

const test1 = cardValues(["3H", "4D", "5S"]);
const test2 = cardValues(["AS", "10S", "10H", "6D", "7D"]);
const test3 = cardValues(["8D", "QS", "2H", "JC", "9C"]);
const test4 = cardValues(["AS", "KS"]);
const test5 = cardValues(["10H", "JH", "QH", "KH", "AH"]);

const check1 = [3, 4, 5];
const check2 = [1, 10, 10, 6, 7];
const check3 = [8, 10, 2, 10, 9];
const check4 = [1, 10];
const check5 = [10, 10, 10, 10, 1];

console.log(`Test1: ${JSON.stringify(test1) === JSON.stringify(check1)}`);
console.log(`Test2: ${JSON.stringify(test2) === JSON.stringify(check2)}`);
console.log(`Test3: ${JSON.stringify(test3) === JSON.stringify(check3)}`);
console.log(`Test4: ${JSON.stringify(test4) === JSON.stringify(check4)}`);
console.log(`Test5: ${JSON.stringify(test5) === JSON.stringify(check5)}`);
